collector/error.go

```diff
@@ -0,0 +1,33 @@
+package collector
+
+import (
+	"github.com/giantswarm/microerror"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var alreadyRegisteredError = &microerror.Error{
+	Kind: "alreadyRegisteredError",
+}
+
+// IsAlreadyRegisteredError asserts alreadyRegisteredError.
+func IsAlreadyRegisteredError(err error) bool {
+	c := microerror.Cause(err)
+	_, ok := c.(prometheus.AlreadyRegisteredError)
+	if ok {
+		return true
+	}
+	if c == alreadyRegisteredError {
+		return true
+	}
```

Comment:
IMO this condition should be the first to check. It's cheap to check and obvious we need it.

Reply:
Is that a rule or something? I don't mind actually. When this is important lets write it down.

Reply:
I don't think there is any rule regarding that. If I'm annoyed enough I'll write something.